File: Solutions/[08]_4.py
def ascending(array):
    now = array[0]
    result = 1
    for i in range(1, len(array)):
        if now < array[i]:
            result += 1
            now = array[i]
    return result

n = int(input())
array = []

for _ in range(n):
    array.append(int(input()))

print(ascending(array))
array.reverse()
print(ascending(array))

File: Solutions/[08]_4.cpp
#include <bits/stdc++.h>

using namespace std;

int n;
vector<int> data;

int ascending() {
	int now = data[0];
	int result = 1;
	for (int i = 1; i < data.size(); i++) {
		if (now < data[i]) {
			result += 1;
			now = data[i];
		}
	}
	return result;
}

int main() {
	cin >> n;
	for (int i = 0; i < n; i++) {
		int x;
		cin >> x;
		data.push_back(x);
	}
	
	cout << ascending() << '\n';
	reverse(data.begin(), data.end());
	cout << ascending() << '\n';
}
